### PR TITLE
Improve HomeKit Name characteristic validation and tests

### PR DESCRIPTION
**Pull request by n0rt0nthec4t**
## :recycle: Current situation

HAP-NodeJS validates the `Name` characteristic using a more restrictive character set than current Home app behaviour. This can result in valid accessory or service names generating warnings unnecessarily.

Examples include names containing punctuation such as `!`, apostrophes (`'`), brackets, slashes, commas, or other commonly accepted characters. In practice, many of these names work correctly in Home but still trigger validation warnings.

There was also limited test coverage around valid vs invalid naming behaviour, particularly for edge cases involving leading/trailing punctuation.

## :bulb: Proposed solution

Refine `Name` characteristic validation to better align with current Home app behaviour.

Changes include:

- Allow a wider set of commonly accepted punctuation characters within names
- Continue enforcing stricter validation for the start and end of names (must begin/end with a letter or number)
- Preserve warnings for clearly problematic names and unsupported symbols
- Expand unit test coverage for accessory and service naming validation

This PR does **not** change `ConfiguredName` behaviour. Existing HomeKit behaviour for `ConfiguredName` remains unchanged, with tests added to confirm it is not subject to `Name` validation rules.

## :gear: Release Notes

Improved HomeKit `Name` characteristic validation to better reflect current Home app behaviour.

Accessory and service names can now include a broader range of commonly accepted punctuation while still warning on invalid leading/trailing characters or unsupported names.

Additional tests were added to validate supported naming patterns and edge cases.

### Testing

Added and updated tests covering:

- Valid accessory and service names containing supported punctuation
- Invalid names with unsupported leading/trailing characters
- Warning generation behaviour for invalid names
- `ConfiguredName` remaining exempt from `Name` validation checks

Covered examples include:

- `Bad ! Name` ✅
- `Bad ' Name` ✅
- `My Bad Switch!` ⚠️
- `'Bad Name` ⚠️

### Reviewer Nudging

Recommended review order:

1. `checkName()` / name validation logic
2. Validation warning messaging
3. Added unit tests in `Accessory.spec.ts`
4. `ConfiguredName` regression test to confirm unchanged behaviour

### CODE REVIEW

File: src/lib/Accessory.spec.ts [hunk 1]
```diff
@@ -384,194 +384,192 @@ describe("Accessory", () => {
 
   describe("Accessory and Service naming checks", () => {
     let consoleWarnSpy: jest.SpyInstance;
-
+
```
Comment (bwp91) [inline src/lib/Accessory.spec.ts]: are you able to please revert all instances like this where just a couple of spaces have been added to empty lines? thanks!

File: src/lib/util/checkName.ts [hunk 2]
```diff
@@ -8,10 +8,11 @@ import { CharacteristicValue, Nullable } from "../../types";
...
 export function checkName(displayName: string, name: string, value: Nullable<CharacteristicValue>): void {
 
-  // Ensure the string starts and ends with a Unicode letter or number and allow any combination of letters, numbers, spaces, and apostrophes in the middle.
-  if (typeof value === "string" && !(new RegExp(/^[\p{L}\p{N}][\p{L}\p{N}\u2019 '.,-]*[\p{L}\p{N}\u2019]$/u)).test(value)) {
-    console.warn("HAP-NodeJS WARNING: The accessory '" + displayName + "' has an invalid '" + name + "' characteristic ('" + value + "'). Please use only " +
-      "alphanumeric, space, and apostrophe characters. Ensure it starts and ends with an alphabetic or numeric character, and avoid emojis. This may prevent " +
-      "the accessory from being added in the Home App or cause unresponsiveness.");
+  // Ensure the string starts and ends with a Unicode letter or number.
+  // Allow letters, numbers, spaces, apostrophes, and common punctuation in the middle.
+  if (typeof value === "string" && !/^[\p{L}\p{N}][\p{L}\p{N}\p{Z}\u2019'&!._:;()/,-]*[\p{L}\p{N}]$/u.test(value)) {
```
Comment (bwp91) [inline src/lib/util/checkName.ts]: i asked claude about this regex, and whilst this mostly looks good, there was just a couple of things that i thought to double check with you:

1. The end of the name lost a character it used to accept
   - Old end class: `[\p{L}\p{N}\u2019]`
   - New end class: `[\p{L}\p{N}]`
   - So a name ending in a curly apostrophe (`’`, `U+2019`) used to be valid, and now warns

2.  `\p{Z}` is much broader than a plain space - it matches every Unicode separator, including:
     - non-breaking space (`U+00A0`)
     - ideographic space used in CJK (`U+3000`)
     - line separator (`U+2028`) and paragraph separator (`U+2029`)
     - several thin/narrow spaces (`U+2000–U+200A`, `U+202F, U+205F`)

    Ideas:
    - If the goal is "any kind of space character", `\p{Zs}` is safer: `\p{Zs}` covers space-like characters only, it leaves out line/paragraph separators
    - If the goal is just "a normal space", keep the old literal (or `\u0020`)

**Comment by n0rt0nthec4t** [comment]
Thanks, Ive updated the naming validation to better align with current Home app behaviour.

This change is intentionally stricter and may introduce warnings for some existing accessories or plugins that previously passed validation. In particular, names ending in apostrophes (including the Unicode curly apostrophe ’) are now warned on, as current versions of the Home app appear to reject names that do not end with a Unicode letter or number.

While this may surface new warnings in some integrations, the intent is to better reflect what HomeKit currently accepts and help plugin/accessory authors avoid naming issues that can prevent accessories from pairing correctly or cause Home app responsiveness problems.

If an accessory or plugin now warns, the expectation is that the name should be updated to comply with current Home app naming requirements.

File: src/lib/util/checkName.ts [hunk 3]
```diff
@@ -8,10 +8,25 @@ import { CharacteristicValue, Nullable } from "../../types";
...
+    !/^[\p{L}\p{N}][\p{L}\p{N}\p{Zs}\u2019'&!._:;()/,-]*[\p{L}\p{N}]$/u.test(value)
+  ) {
+    console.warn(
+      "HAP-NodeJS WARNING: The accessory '" + displayName +
+      "' has an invalid '" + name +
+      "' characteristic ('" + value +
+      "'). Please ensure the name starts and ends with a letter or number. " +
+      "Only letters, numbers, spaces, apostrophes, and common punctuation " +
+      "are supported. Avoid emojis or unsupported symbols. This may prevent " +
+      "the accessory from being added in the Home App or cause unresponsiveness."
```
Comment (bwp91) [inline src/lib/util/checkName.ts]: just one lint issue @n0rt0nthec4t

**Review by bwp91 — APPROVED** [review]
thanks for this! and for making those changes
